jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/TransitNodeRoutingPrecomputation.java

```diff
@@ -0,0 +1,1214 @@
+                                        <= manyToManyShortestPaths.getWeight(v2, v)) {
+                                    result.add(v2);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            return result;
+        }
+    }
+
+    /**
+     * Provides API to build a {@code LocalityFilter} object.
+     */
+    private class LocalityFilterBuilder {
+        /**
+         * Visited Voronoi cells by a forward {@code ContractionHierarchyBFS} search.
+         */
+        private List<Set<Integer>> visitedForwardVoronoiCells;
+        /**
+         * Visited Voronoi cells by a backward {@code ContractionHierarchyBFS} search.
+         */
+        private List<Set<Integer>> visitedBackwardVoronoiCells;
+
+        /**
+         * Constructs an instance for the given {@code numberOfVertices}.
+         *
+         * @param numberOfVertices number of vertices in graph
+         */
+        public LocalityFilterBuilder(int numberOfVertices) {
+            this.visitedForwardVoronoiCells = new ArrayList<>(numberOfVertices);
+            this.visitedBackwardVoronoiCells = new ArrayList<>(numberOfVertices);
+            for (int i = 0; i < numberOfVertices; ++i) {
+                visitedForwardVoronoiCells.add(null);
+                visitedBackwardVoronoiCells.add(null);
+            }
+        }
+
+
+        /**
+         * Adds {@code visitedVoronoiCells} to this builder in the forward
+         * direction for {@code vertex}.
+         *
+         * @param vertex              vertex
+         * @param visitedVoronoiCells visited Voronoi cells
+         */
+        public void addForwardVisitedVoronoiCells(ContractionVertex<V> vertex, Set<Integer> visitedVoronoiCells) {
+            this.visitedForwardVoronoiCells.set(vertex.vertexId, visitedVoronoiCells);
+        }
+
+        /**
+         * Adds {@code visitedVoronoiCells} to this builder in the backward
+         * direction for {@code vertex}.
+         *
+         * @param vertex              vertex
+         * @param visitedVoronoiCells visited Voronoi cells
+         */
+        public void addBackwardVisitedVoronoiCells(ContractionVertex<V> vertex, Set<Integer> visitedVoronoiCells) {
+            this.visitedBackwardVoronoiCells.set(vertex.vertexId, visitedVoronoiCells);
+        }
+
+
+        /**
+         * Builds an instance of {@code LocalityFiler} using {@code visitedForwardVoronoiCells}
+         * and {@code visitedBackwardVoronoiCells}.
+         *
+         * @return locality filter
+         */
+        public LocalityFiler<V> buildLocalityFilter() {
+            return new LocalityFiler<>(contractionMapping, visitedForwardVoronoiCells, visitedBackwardVoronoiCells);
+        }
+    }
+
+
+    /**
+     * Return type of this algorithm. Contains {@code contractionHierarchy},
```

Comment:
This class required significant Javadoc improvement. Specifically, please, include
* a general information of what this class contains logically, not the private variable names. 
* preferably, how this information can be used